docs/cookbook/using-docker.md

```diff
@@ -0,0 +1,343 @@
+
+First, make sure your `package-lock.json` file is up to date by running `npm update` on your project repo. Then, with the container down, run:
+
+```bash
+docker compose build
+```
+
+And then run:
+
+```bash
+docker compose up
+```
+
+### Summary
+While in this example, our project is still being hosted locally, any of these commands can be issued on a server that supports Docker once your project is deployed.
+
+Right now, our Dockerized container is limited to a single server. For simple, low-traffic sites this could be fine. However, if we want to scale our site over several servers and add a load balancer like Nginix, we need to add support for cloud storage and a cloud database. Fortunately, Apostrophe makes this relatively easy.
+
+## Using AWS S3 services
+If you aren't hosting your project on a single server, you will need to have a different uploaded asset storage method. Typically this is a service like Amazon Web Services S3 or another similar service. Apostrophe is set up to easily use S3 services by adding environment variables. You can read more in the [documentation](https://v3.docs.apostrophecms.org/reference/modules/uploadfs.html#s3-storage-options). We can take advantage of this in Docker by expanding our `docker-compose.yml` and `.env` files.
+
+### Changing the `docker-compose.yaml` file
+In order to pass the environment variables into our project container we just need to add them inside the `environment:` key. If we are using S3 services at Amazon, we need to add four variables: `APOS_S3_REGION`, `APOS_S3_BUCKET`, `APOS_S3_KEY`, and `APOS_S3_SECRET`. For other S3-type storage solutions, such as [filebase](https://filebase.com/), you will also want to set the `APOS_S3_ENDPOINT` variable. For AWS, your `environment:` section should now look like this:
+
+<AposCodeBlock>
+
+```bash
+…
+    environment:
+      - NODE_ENV
+      - APOS_MONGODB_URI
+      - APOS_CLUSTER_PROCESSES
+      - APOS_S3_REGION
+      - APOS_S3_BUCKET
+      - APOS_S3_KEY
+      - APOS_S3_SECRET
+…
+```
+
+<template v-slot:caption>
+  docker-compose.yaml
+</template>
+
+</AposCodeBlock>
+
+### Changing the `.env` file
+Next, the `.env` file should be modified to contain values for each of the new environment variables. Each will get populated with values specific to your S3 buckets. Again, add the `APOS_S3_ENDPOINT` with value if using a service not hosted by AWS.
+
+<AposCodeBlock>
+
+```sh
+NODE_ENV=production
+APOS_MONGODB_URI=mongodb://db:27017/apostrophe
+APOS_S3_REGION=<your region>
+APOS_S3_BUCKET=<your bucket name>
+APOS_S3_KEY=<account key>
+APOS_S3_SECRET=<account secret>
+```
+
+<template v-slot:caption>
+  .env
+</template>
+
+</AposCodeBlock>
+
+In addition to these changes, you will also likely have to change your `@apostrophecms/uploadfs` options for accessing the bucket. For example:
+
+<AposCodeBlock>
+
+```javascript
+module.exports = {
+  options: {
+    uploadfs: {
+      bucketObjectsACL: 'bucket-owner-full-control'
```

Comment:
Is this option really necessary?

on the other hand, the `secure` option *is* necessary for production — see the latest in discord. We need to start enabling that option by default in A3 so we don't have to talk about it.

Reply:
I did see that in the Discord. Amazon's policy to make new buckets non-public by default requires that you either edit the ACL at Amazon, but this at least allows local write, but not read. Getting into permissions policy could be a bit of a rabbit hole.

Reply:
I hedged, but I'm willing to just get rid of this if you think it would be better.

Reply:
I've never set this option. What I've done instead is make sure that in the AWS S3 control panel, things are set up like this (fusillade of screenshots follow)

<img width="1418" alt="Screen Shot 2022-09-21 at 8 51 39 AM" src="https://user-images.githubusercontent.com/32125/191508789-6690bbcc-73f5-4ea8-9370-3bb31f7ffe7d.png">
<img width="921" alt="Screen Shot 2022-09-21 at 8 51 45 AM" src="https://user-images.githubusercontent.com/32125/191508791-b8b71c3c-d532-41ac-b599-d8c99bb807e9.png">
<img width="1425" alt="Screen Shot 2022-09-21 at 8 52 08 AM" src="https://user-images.githubusercontent.com/32125/191508795-f89fa3c5-30b4-4912-96a6-876a4960f921.png">
<img width="901" alt="Screen Shot 2022-09-21 at 8 52 13 AM" src="https://user-images.githubusercontent.com/32125/191508797-49436d11-5b30-4803-a2bb-6bf3a4f808a8.png">

Reply:
Added a bit of text and deleted code example.

Reply:
Images and text added.